flatsurf/geometry/matrix_2x2.py

```diff
@@ -183,62 +110,6 @@ def similarity_from_vectors(u,v):
     return matrix([[cos_uv, -sin_uv],[sin_uv, cos_uv]])
 
 
-def rotation_matrix_angle(r, check=False):
-    r"""
-    Return the angle of the rotation matrix ``r`` divided by ``2 pi``.
-
-    EXAMPLES::
-
-        sage: from flatsurf.geometry.matrix_2x2 import rotation_matrix_angle
-
-        sage: def rot_matrix(p, q):
-        ....:     z = QQbar.zeta(q) ** p
-        ....:     c = z.real()
-        ....:     s = z.imag()
-        ....:     return matrix(AA, 2, [c,-s,s,c])
-        sage: [rotation_matrix_angle(rot_matrix(i, 5)) for i in range(1,5)]
-        [1/5, 2/5, 3/5, 4/5]
-        sage: [rotation_matrix_angle(rot_matrix(i, 5)) for i in range(1,5)]
-        [1/5, 2/5, 3/5, 4/5]
-        sage: [rotation_matrix_angle(rot_matrix(i,7)) for i in range(1,7)]
-        [1/7, 2/7, 3/7, 4/7, 5/7, 6/7]
-
-    Some random tests::
-
-        sage: for _ in range(100):
-        ....:     r = QQ.random_element(x=0,y=500)
-        ....:     r -= r.floor()
-        ....:     m = rot_matrix(r.numerator(), r.denominator())
-        ....:     assert rotation_matrix_angle(m) == r
-
-    .. NOTE::
-
-        This is using floating point arithmetic and might be wrong.
-    """
-    e0,e1 = r.change_ring(CDF).eigenvalues()
-    m0 = (e0.log() / 2 / CDF.pi()).imag()
-    m1 = (e1.log() / 2 / CDF.pi()).imag()
-    r0 = RR(m0).nearby_rational(max_denominator=10000)
-    r1 = RR(m1).nearby_rational(max_denominator=10000)
-    if r0 != -r1:
-        raise RuntimeError
```

Comment:
This assertion sometimes failed, see #151.

Reply:
It currently fails on master!